ecosystem/typescript/sdk_v2/src/types/account_address.ts

```diff
@@ -0,0 +1,241 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+import { bytesToHex, hexToBytes } from "@noble/hashes/utils";
+import { Serializer, Deserializer, Bytes } from "../bcs";
+
+// TODO: Get feedback on these error messages, ideally they can be used directly in
+// dapps and wallets. The enum is meant to make it possible for daps to user their
+// own error messages if they want, but the defaults should still be good, and they're
+// not great right now.
+
+// These options influence fromStr and the isValid* functions. You may use them to make
+// the parsing of those functions strictier. Not setting these options is the best way
+// to comply with AIP 40 but while migrating it may be helpful to be able to enforce
+// stricter behavior. If we agree to make these two enabled by default, I can update
+// the AIP to say that this is what people SHOULD do but they MAY opt in to accepting
+// addresses more leniently in specifically these two ways.
+export type ValidityOptions = {
+  // Only addresses in the long form (64 characters + optionally a leading 0x) will be
+  // considered valid unless the address is special.
+  requireLongFormUnlessSpecial?: boolean;
+
+  // Only addresses in the long form (64 characters + optionally a leading 0x) will be
+  // considered valid, even if the address is special.
+  requireLongForm?: boolean;
+
+  // If set, the address string will be considered invalid if there is no leading 0x.
+  requireLeadingZeroX?: boolean;
+};
+
+// TODO: As part of reviewing this we should discuss these options. I'm partially
+// inclined to say that requireLongFormUnlessSpecial and requireLeadingZeroX should
+// actually be enabled by default. Maayan and Greg to provide their thoughts as well as
+// wallet and dapp folks.
+export const defaultValidityOptions: ValidityOptions = {
+  requireLongFormUnlessSpecial: false,
+  requireLongForm: false,
+  requireLeadingZeroX: false,
+};
+
+export enum AddressInvalidReason {
+  INCORRECT_NUMBER_OF_BYTES = "incorrect_number of bytes",
+  ADDRESS_TOO_SHORT = "address_too_short",
```

Comment:
maybe?
```suggestion
  ADDRESS_TOO_SHORT = "address too short should be between 1 to 64",
```

Reply:
So this isn't really meant to be the explanation, that comes from within the function. The value for this enum could just as easily be a number rather than a string, which maybe I'll do.